rmmxapianapi.cc

```diff
@@ -523,6 +523,58 @@ extern "C" {
       return numfolders;
     }
 
+    // returns a pair: [total, unread] in `results[]`
+    int EMSCRIPTEN_KEEPALIVE getFolderMessageCounts(const char *folderName, int results[]) {
+        if (!dbc) return 0;
+
+        Xapian::QueryParser queryparser;
+        queryparser.set_database(dbc->db);
+        queryparser.add_boolean_prefix("folder", "XFOLDER:");
+        queryparser.add_boolean_prefix("flag", "XF");
+
+        string queryString = "folder:\"";
+        queryString += folderName;
+        queryString += "\"";
+
+        try {
+            {
+                Xapian::Enquire enquire(dbc->db);
```

Comment:
@petersalomonsen it feels like this should be doable with an `mset.get_termfreq("XFseen")` on the full query below here, but for some reason that always yields exactly the same number as the total number of messages. Anything obvious I'm missing?

Reply:
sorry for the delay @tadzik, just saw your message. From the [docs](https://xapian.org/docs/apidoc/html/classXapian_1_1MSet.html#aa869331afad28c2d815d0553198c4914) it's not clear to me if `termfreq` returns number of occurrences in the resultset or the entire index, maybe @ojwb can tell.

Anyway it would be good if you created a test case in https://github.com/runbox/runbox-searchindex/blob/master/ts/xapian/xapian.test.ts so that it would be easier for others to play with this new function. I found it useful/easier with a test-driven approach when working on this library (there's also a watcher that pick up changes in the ts-files and re-run the tests).

Reply:
It's the number of documents from the entire index that the term occurs in.

It should give exactly the same number as `db.get_termfreq("XFseen")` would, but if the term was in the query and contributed to the relevance weight then it returns the cached value rather than asking the database, which potentially saves having to read from disk.

Reply:
Good to know, thanks. Is there anything else I could use here to avoid doing two separate queries?

Reply:
Can't think of any better solution right now. Seems like you have to query it twice.

Would it be an idea with a more generic method to get results count for any query ( and instead call it with and without the `seen` flag from Javascript )? Then it could be used for other kinds of result counts too. Wouldn't increase performance, but give more reusability.

Reply:
I finally got to clarifying the docs here: xapian/xapian@092d91523991696152c6b48ce546c4b3b55ac9f3